crates/dojo-indexer/src/stream.rs

```diff
@@ -0,0 +1,92 @@
+use std::str::FromStr;
+
+use apibara_client_protos::pb::starknet::v1alpha2::Filter;
+use apibara_client_protos::pb::stream::v1alpha2::stream_client::StreamClient;
+use apibara_client_protos::pb::stream::v1alpha2::stream_data_response::Message;
+use apibara_client_protos::pb::stream::v1alpha2::{
+    Cursor, Data, StreamDataRequest, StreamDataResponse,
+};
+use futures_util::Stream;
+use log::debug;
+use tokio::sync::mpsc;
+use tonic::transport::{Channel, Endpoint};
+
+pub enum Chain {
+    AlphaMainnet,
+    AlphaGoerli,
+    AlphaGoerli2,
+}
+
+impl From<Chain> for &'static str {
+    fn from(chain: Chain) -> Self {
+        match chain {
+            Chain::AlphaMainnet => "https://mainnet.starknet.a5a.ch:443",
+            Chain::AlphaGoerli => "https://goerli.starknet.a5a.ch:443",
+            Chain::AlphaGoerli2 => "https://goerli2.starknet.a5a.ch:443",
+        }
+    }
+}
+
+pub struct ApibaraClient {
+    stream: StreamClient<Channel>,
+    stream_id: u64,
+    response_stream: Option<tonic::codec::Streaming<StreamDataResponse>>,
+    sender: Option<mpsc::Sender<StreamDataRequest>>,
+}
+
+impl ApibaraClient {
+    pub async fn new(path: &String) -> Result<Self, tonic::transport::Error> {
+        println!("{}", path);
+        let channel = Endpoint::from_str(path)?.connect().await?;
+
+        debug!("Connected to Apibara server {:?}", channel);
+        let stream = StreamClient::new(channel);
+        Ok(Self { stream, stream_id: 0, response_stream: None, sender: None })
+    }
+
+    #[allow(clippy::needless_lifetimes)]
+    pub async fn request_data<'x>(
+        &'x mut self,
+        filter: Filter,
+    ) -> Result<impl Stream<Item = Result<Option<Data>, tonic::Status>> + 'x, tonic::Status> {
+        self.stream_id += 1;
+        let request = StreamDataRequest {
+            stream_id: Some(self.stream_id),
+            batch_size: Some(1),
+            starting_cursor: Some(Cursor { order_key: 1, unique_key: vec![] }),
```

Comment:
If you set `starting_cursor` to `None` it will stream from genesis.

Reply:
hm we prob want the start block as part of the world config